src/plugins/fcrypt/fcrypt.c

```diff
@@ -230,15 +234,29 @@ static int fcryptGpgCallAndCleanup (Key * parentKey, KeySet * pluginConfig, char
 		// gpg call returned success, overwrite the original file with the gpg payload data
 		if (rename (tmpFile, keyString (parentKey)) != 0)
 		{
-			ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Renaming file %s to %s failed. Reason: %s", tmpFile, keyString (parentKey),
-						     strerror (errno));
-			result = -1;
+			// if rename failed we can still try to copy the file content manually
+			lseek (tmpFileFd, 0, SEEK_SET);
+			lseek (parentKeyFd, 0, SEEK_SET);
+			readCount = read (tmpFileFd, buffer, sizeof (buffer));
+			while (readCount > 0)
+			{
+				writeCount = write (parentKeyFd, buffer, readCount);
+				if (writeCount != readCount)
```

Comment:
This check is too strict, it is perfectly okay when write does not write everything at once but only a junk. Be beware of not introducing silent errors here.